Fields with a const specifier result in wrong key names

```cpp
struct tst {
   const int foo;
}

tst b{1};
std::string payload;
auto err = glz::write_json(b, payload);
```
I would expect this code to work and yield `{"foo":1}`. But it yields `{"foo)}]":1}`. Removing the `const` specifier resolves this problem.


<details><summary> Thankfully you get a nice-ish compiler error (Edit: unrelated) </summary>

```
[ 33%] Built target jsonschema_test
[ 50%] Building CXX object tests/json_test/CMakeFiles/json_test.dir/json_test.cpp.o
[ 66%] Building CXX object tests/json_test/CMakeFiles/json_test_non_null.dir/json_test.cpp.o
In file included from /home/max/desk/glaze/include/glaze/json/json_ptr.hpp:11,
                 from /home/max/desk/glaze/include/glaze/core/ptr.hpp:9,
                 from /home/max/desk/glaze/include/glaze/beve/ptr.hpp:8,
                 from /home/max/desk/glaze/include/glaze/beve.hpp:7,
                 from /home/max/desk/glaze/include/glaze/glaze.hpp:35,
                 from /home/max/desk/glaze/include/glaze/api/impl.hpp:24,
                 from /home/max/desk/glaze/tests/json_test/json_test.cpp:25:
In static member function ‘static void glz::detail::from<10, T>::op(auto:619&&, auto:620&&, auto:621&&,
 auto:622&&) [with auto Options = glz::opts{10, '\001', '\000', '\001', '\001', '\001', '\000', '\000',
 ' ', 3, '\001', '\000', '\000', '\001', '\000', '\000', '\000', '\000', 0, glz::float_precision::full,
 '\000', '\000', '\000', '\000', '\000', '\000', '\001', '\000', '\001', '\001', 32}; auto:619 = std::v
ariant<A1>&; auto:620 = glz::context&; auto:621 = const char*&; auto:622 = const char*&; T = std::varia
nt<A1>]’,
    inlined from ‘static void glz::detail::read<10>::op(T&&, Ctx&&, It0&&, It1&&) [with auto Opts = glz
::opts{10, '\001', '\000', '\001', '\001', '\001', '\000', '\000', ' ', 3, '\001', '\000', '\000', '\00
1', '\000', '\000', '\000', '\000', 0, glz::float_precision::full, '\000', '\000', '\000', '\000', '\00
0', '\000', '\001', '\000', '\001', '\001', 32}; T = std::variant<A1>&; Ctx = glz::context&; It0 = cons
t char*&; It1 = const char*&]’ at /home/max/desk/glaze/include/glaze/json/read.hpp:74:48,
    inlined from ‘glz::error_ctx glz::read(T&, auto:219&&, auto:220&&) [with opts Opts = opts{10, '\001
', '\000', '\001', '\001', '\001', '\000', '\000', ' ', 3, '\001', '\000', '\000', '\001', '\000', '\00
0', '\000', '\000', 0, glz::float_precision::full, '\000', '\000', '\000', '\000', '\000', '\000', '\00
1', '\000', '\001', '\001', 0}; T = std::variant<A1>; auto:219 = std::__cxx11::basic_string<char>&; aut
o:220 = context&]’ at /home/max/desk/glaze/include/glaze/core/read.hpp:60:70,
    inlined from ‘glz::expected<T, glz::error_ctx> glz::read_json(Buffer&&) [with T = std::variant<A1>;
 Buffer = std::__cxx11::basic_string<char>&]’ at /home/max/desk/glaze/include/glaze/json/read.hpp:2749:
40:
/home/max/desk/glaze/include/glaze/json/read.hpp:2275:70: warning: array subscript 8 is outside array b
ounds of ‘std::array<std::variant<A1>, 1> [1]’ [-Warray-bounds=]
 2275 |                                     if (value.index() != type_index) value = runtime_variant_ma
p<T>()[type_index];
      |                                                                      ^~~~~
/home/max/desk/glaze/include/glaze/json/read.hpp: In function ‘glz::expected<T, glz::error_ctx> glz::re
ad_json(Buffer&&) [with T = std::variant<A1>; Buffer = std::__cxx11::basic_string<char>&]’:
/home/max/desk/glaze/include/glaze/json/read.hpp:2275:100: note: at offset 8 into object ‘<anonymous>’ 
of size 8
 2275 |                     if (value.index() != type_index) value = runtime_variant_map<T>()[type_inde
x];
      |                                                              ~~~~~~~~~~~~~~~~~~~~~~^~

In file included from /home/max/desk/glaze/include/glaze/json/json_ptr.hpp:11,
                 from /home/max/desk/glaze/include/glaze/core/ptr.hpp:9,
                 from /home/max/desk/glaze/include/glaze/beve/ptr.hpp:8,
                 from /home/max/desk/glaze/include/glaze/beve.hpp:7,
                 from /home/max/desk/glaze/include/glaze/glaze.hpp:35,
                 from /home/max/desk/glaze/include/glaze/api/impl.hpp:24,
                 from /home/max/desk/glaze/tests/json_test/json_test.cpp:25:
In static member function ‘static void glz::detail::from<10, T>::op(auto:619&&, auto:620&&, auto:621&&,
 auto:622&&) [with auto Options = glz::opts{10, '\000', '\000', '\001', '\001', '\001', '\000', '\000',
 ' ', 3, '\001', '\000', '\000', '\001', '\000', '\000', '\000', '\000', 0, glz::float_precision::full,
 '\000', '\000', '\000', '\000', '\000', '\000', '\001', '\000', '\001', '\001', 32}; auto:619 = std::v
ariant<A1>&; auto:620 = glz::context&; auto:621 = const char*&; auto:622 = const char*&; T = std::varia
nt<A1>]’,
    inlined from ‘static void glz::detail::read<10>::op(T&&, Ctx&&, It0&&, It1&&) [with auto Opts = glz
::opts{10, '\000', '\000', '\001', '\001', '\001', '\000', '\000', ' ', 3, '\001', '\000', '\000', '\00
1', '\000', '\000', '\000', '\000', 0, glz::float_precision::full, '\000', '\000', '\000', '\000', '\00
0', '\000', '\001', '\000', '\001', '\001', 32}; T = std::variant<A1>&; Ctx = glz::context&; It0 = cons
t char*&; It1 = const char*&]’ at /home/max/desk/glaze/include/glaze/json/read.hpp:74:48,
    inlined from ‘glz::error_ctx glz::read(T&, auto:219&&, auto:220&&) [with opts Opts = opts{10, '\000
', '\000', '\001', '\001', '\001', '\000', '\000', ' ', 3, '\001', '\000', '\000', '\001', '\000', '\00
0', '\000', '\000', 0, glz::float_precision::full, '\000', '\000', '\000', '\000', '\000', '\000', '\00
1', '\000', '\001', '\001', 0}; T = std::variant<A1>; auto:219 = std::__cxx11::basic_string<char>&; aut
o:220 = context&]’ at /home/max/desk/glaze/include/glaze/core/read.hpp:60:70,
    inlined from ‘glz::expected<T, glz::error_ctx> glz::read_json(Buffer&&) [with T = std::variant<A1>;
 Buffer = std::__cxx11::basic_string<char>&]’ at /home/max/desk/glaze/include/glaze/json/read.hpp:2749:
40:
/home/max/desk/glaze/include/glaze/json/read.hpp:2275:70: warning: array subscript 8 is outside array b
ounds of ‘std::array<std::variant<A1>, 1> [1]’ [-Warray-bounds=]
 2275 |                                     if (value.index() != type_index) value = runtime_variant_ma
p<T>()[type_index];
      |                                                                      ^~~~~
/home/max/desk/glaze/include/glaze/json/read.hpp: In function ‘glz::expected<T, glz::error_ctx> glz::re
ad_json(Buffer&&) [with T = std::variant<A1>; Buffer = std::__cxx11::basic_string<char>&]’:
/home/max/desk/glaze/include/glaze/json/read.hpp:2275:100: note: at offset 8 into object ‘<anonymous>’ 
of size 8
 2275 |                                     if (value.index() != type_index) value = runtime_variant_ma
p<T>()[type_index];
      |                                                                              ~~~~~~~~~~~~~~~~~~
~~~~^~
In file included from /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-202411
16/bitset:49,
                 from /home/max/desk/glaze/tests/json_test/json_test.cpp:5:
In static member function ‘static constexpr _Up* std::__copy_move<_IsMove, true, std::random_access_ite
rator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = long unsigned int; _Up = long unsigned int; bool _Is
Move = false]’,
    inlined from ‘constexpr _OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = lo
ng unsigned int*; _OI = long unsigned int*]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-2024
1116/include/c++/14-20241116/bits/stl_algobase.h:521:30,
    inlined from ‘constexpr _OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = lo
ng unsigned int*; _OI = long unsigned int*]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-2024
1116/include/c++/14-20241116/bits/stl_algobase.h:548:42,
    inlined from ‘constexpr _OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = lon
g unsigned int*; _OI = long unsigned int*]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241
116/include/c++/14-20241116/bits/stl_algobase.h:555:31,
    inlined from ‘constexpr _OI std::copy(_II, _II, _OI) [with _II = long unsigned int*; _OI = long uns
igned int*]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bit
s/stl_algobase.h:651:7,
    inlined from ‘constexpr std::vector<bool, _Alloc>::iterator std::vector<bool, _Alloc>::_M_copy_alig
ned(const_iterator, const_iterator, iterator) [with _Alloc = std::allocator<bool>]’ at /nix/store/l89iq
c7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/stl_bvector.h:1342:28,
    inlined from ‘constexpr void std::vector<bool, _Alloc>::_M_insert_range(iterator, _ForwardIterator,
 _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = const bool*; _Alloc = std::alloc
ator<bool>]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bit
s/vector.tcc:1124:33,
    inlined from ‘constexpr std::vector<bool, _Alloc>::iterator std::vector<bool, _Alloc>::insert(const
_iterator, _InputIterator, _InputIterator) [with _InputIterator = const bool*; <template-parameter-2-2>
 = void; _Alloc = std::allocator<bool>]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116
/include/c++/14-20241116/bits/stl_bvector.h:1219:19,
    inlined from ‘constexpr void std::vector<bool, _Alloc>::_M_assign_aux(_ForwardIterator, _ForwardIte
rator, std::forward_iterator_tag) [with _ForwardIterator = const bool*; _Alloc = std::allocator<bool>]’
 at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/stl_bvecto
r.h:1479:14,
    inlined from ‘constexpr void std::vector<bool, _Alloc>::assign(_InputIterator, _InputIterator) [wit
h _InputIterator = const bool*; <template-parameter-2-2> = void; _Alloc = std::allocator<bool>]’ at /ni
x/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/stl_bvector.h:974
:17,
    inlined from ‘constexpr std::vector<bool, _Alloc>& std::vector<bool, _Alloc>::operator=(std::initia
lizer_list<bool>) [with _Alloc = std::allocator<bool>]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-
gcc-14-20241116/include/c++/14-20241116/bits/stl_bvector.h:954:14,
    inlined from ‘<lambda()>::<lambda()>’ at /home/max/desk/glaze/tests/json_test/json_test.cpp:1483:6
:
/nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/stl_algobase.h
:452:30: warning: ‘void* __builtin_memmove(void*, const void*, long unsigned int)’ forming offset 8 is 
out of the bounds [0, 8] [-Warray-bounds=]
  452 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-202411
16/bitset:49,
                 from /home/max/desk/glaze/tests/json_test/json_test.cpp:5:
In static member function ‘static constexpr _Up* std::__copy_move<_IsMove, true, std::random_access_ite
rator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = long unsigned int; _Up = long unsigned int; bool _Is
Move = false]’,
    inlined from ‘constexpr _OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = lo
ng unsigned int*; _OI = long unsigned int*]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-2024
1116/include/c++/14-20241116/bits/stl_algobase.h:521:30,
    inlined from ‘constexpr _OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = lo
ng unsigned int*; _OI = long unsigned int*]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-2024
1116/include/c++/14-20241116/bits/stl_algobase.h:548:42,
    inlined from ‘constexpr _OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = lon
g unsigned int*; _OI = long unsigned int*]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241
116/include/c++/14-20241116/bits/stl_algobase.h:555:31,
    inlined from ‘constexpr _OI std::copy(_II, _II, _OI) [with _II = long unsigned int*; _OI = long uns
igned int*]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bit
s/stl_algobase.h:651:7,
    inlined from ‘constexpr std::vector<bool, _Alloc>::iterator std::vector<bool, _Alloc>::_M_copy_alig
ned(const_iterator, const_iterator, iterator) [with _Alloc = std::allocator<bool>]’ at /nix/store/l89iq
c7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/stl_bvector.h:1342:28,
    inlined from ‘constexpr void std::vector<bool, _Alloc>::_M_insert_range(iterator, _ForwardIterator,
 _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = const bool*; _Alloc = std::alloc
ator<bool>]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bit
s/vector.tcc:1124:33,
    inlined from ‘constexpr std::vector<bool, _Alloc>::iterator std::vector<bool, _Alloc>::insert(const
_iterator, _InputIterator, _InputIterator) [with _InputIterator = const bool*; <template-parameter-2-2>
 = void; _Alloc = std::allocator<bool>]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116
/include/c++/14-20241116/bits/stl_bvector.h:1219:19,
    inlined from ‘constexpr void std::vector<bool, _Alloc>::_M_assign_aux(_ForwardIterator, _ForwardIte
rator, std::forward_iterator_tag) [with _ForwardIterator = const bool*; _Alloc = std::allocator<bool>]’
 at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/stl_bvecto
r.h:1479:14,
    inlined from ‘constexpr void std::vector<bool, _Alloc>::assign(_InputIterator, _InputIterator) [wit
h _InputIterator = const bool*; <template-parameter-2-2> = void; _Alloc = std::allocator<bool>]’ at /ni
x/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/stl_bvector.h:974
:17,
    inlined from ‘constexpr std::vector<bool, _Alloc>& std::vector<bool, _Alloc>::operator=(std::initia
lizer_list<bool>) [with _Alloc = std::allocator<bool>]’ at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-
gcc-14-20241116/include/c++/14-20241116/bits/stl_bvector.h:954:14,
    inlined from ‘<lambda()>::<lambda()>’ at /home/max/desk/glaze/tests/json_test/json_test.cpp:1483:6
:
/nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/stl_algobase.h
:452:30: warning: ‘void* __builtin_memmove(void*, const void*, long unsigned int)’ forming offset 8 is 
out of the bounds [0, 8] [-Warray-bounds=]
  452 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

</details>

I think this is a regression, because it broke after an update.
I will try to bisect later today. 
Bisected to https://github.com/stephenberry/glaze/commit/47d20fe64968e0fb257e5007b3e2f9f5c03f9101

--- a/include/glaze/reflection/get_name.hpp
+++ b/include/glaze/reflection/get_name.hpp
@@ -21,8 +21,10 @@
 namespace glz::detail
 {
    // Do not const qualify this value to avoid duplicate `to_tie` template instantiations with rest of Glaze
+   // Temporary fix: const qualify it despite. Caused issues with reflect begin/end indices
+   // See https://github.com/stephenberry/glaze/issues/1568
    template <class T>
-   extern T external;
+   extern const T external;
 
    // using const char* simplifies the complier's output and should improve compile times
    template <auto Ptr>

